core/src/main/java/com/microsoft/applicationinsights/internal/statsbeat/AttachStatsbeat.java

```diff
@@ -44,6 +45,11 @@
 
     @Override
     protected void send() {
+        // WEBSITE_HOSTNAME is lazily set in Linux Consumption Plan.
+        if (Strings.isNullOrEmpty(resourceProviderId)) {
+            resourceProviderId = System.getenv("WEBSITE_HOSTNAME");
```

Comment:
effectively the same, just so that we only have one source of truth for initializing resourceProviderId
```suggestion
            resourceProviderId = initResourceProviderId(CustomDimensions.get().getResourceProvider(), null);
```